### ISSUE TITLE
Getting error:  Option "allow_alias" unknown.

### ISSUE THREAD

**Issue by potato20** [google/protobuf#2292]
Using latest version, 3.1.0, I'm trying to use `protoc` to compile this

https://github.com/SteamRE/SteamKit/tree/master/Resources/Protobufs

into Ruby. I simply clone into my machine, cd into the directory, then run the command

`protoc --proto_path=./csgo --proto_path=./ --ruby_out=./outlib csgo/**/*.proto`

and i get this

```
[libprotobuf WARNING google/protobuf/compiler/parser.cc:546] No syntax specified for the proto file: base_gcmessages.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
[libprotobuf WARNING google/protobuf/compiler/parser.cc:546] No syntax specified for the proto file: steammessages.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
[libprotobuf WARNING google/protobuf/compiler/parser.cc:546] No syntax specified for the proto file: google/protobuf/descriptor.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
[libprotobuf WARNING google/protobuf/compiler/parser.cc:546] No syntax specified for the proto file: cstrike15_gcmessages.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
[libprotobuf WARNING google/protobuf/compiler/parser.cc:546] No syntax specified for the proto file: engine_gcmessages.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
[libprotobuf WARNING google/protobuf/compiler/parser.cc:546] No syntax specified for the proto file: cstrike15_usermessages.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
[libprotobuf WARNING google/protobuf/compiler/parser.cc:546] No syntax specified for the proto file: netmessages.proto. Please use 'syntax = "proto2";' or 'syntax = "proto3";' to specify a syntax version. (Defaulted to proto2 syntax.)
netmessages.proto:38:16: Option "allow_alias" unknown.
cstrike15_usermessages.proto: Import "netmessages.proto" was not found or had errors.
cstrike15_usermessages.proto:119:18: ".CMsgVector2D" is not defined.
cstrike15_usermessages.proto:120:18: ".CMsgRGBA" is not defined.
cstrike15_usermessages.proto:121:18: ".CMsgRGBA" is not defined.
cstrike15_usermessages.proto:141:18: ".CMsgRGBA" is not defined.
cstrike15_usermessages.proto:185:18: ".CMsgVector" is not defined.
cstrike15_usermessages.proto: warning: Import google/protobuf/descriptor.proto but not used.

```

`protoc --version`

```
libprotoc 3.1.0
```

**Comment by potato20** [comment]
I'm also getting this on librotoc 2.6.1

**Comment by xfxyjwf** [comment]
I think the problem should be fixed if you use this command line:
```bash
$ protoc --proto_path=./csgo --ruby_out=./outlib csgo/**/*.proto
```
(i.e., remove "--proto_path=./").

**Comment by potato20** [comment]
@xfxyjwf if I do that, I get this

```
google/protobuf/descriptor.proto: File not found.
steammessages.proto: Import "google/protobuf/descriptor.proto" was not found or had errors.
steammessages.proto:6:8: ".google.protobuf.FieldOptions" is not defined.
steammessages.proto:10:8: ".google.protobuf.MessageOptions" is not defined.
steammessages.proto: ".google.protobuf.MessageOptions" is not defined.
base_gcmessages.proto: Import "steammessages.proto" was not found or had errors.
```

which is why I was originally using `--proto_path=./`.

If I actually copy the `google` directory into the `csgo` directory and run your command, things work as normal.

However I dont see how or why the `allow_alias` error I reported would even be an issue with the `proto_path`s I specified.

**Comment by potato20** [comment]
Btw I'm using the [protobuf](https://github.com/ruby-protobuf/protobuf) if it's relevant.

**Comment by xfxyjwf** [comment]
For some reason the protos you are trying to compile has a forked version of google/protobuf/descriptor.proto:
https://github.com/SteamRE/SteamKit/blob/master/Resources/Protobufs/google/protobuf/descriptor.proto

If you add --proto_path=./, protocol compiler will include this forked version instead of the one from protobuf installation. So you can't add "--proto_path=./".

Have you tried the command I suggested with protoc 3.1? It won't work with protoc 2.6.1.

**Comment by xfxyjwf** [comment]
Also, make sure you have installed the protoc properly with all the well-known proto files (which contains google/protobuf/descriptor.proto).

**Comment by potato20** [comment]
I tried your command using protoc 3.1.0. Although I had to insert `syntax = "proto3";` to the top of all files, I didn't compile for the following errors

```
base_gcmessages.proto:36:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:37:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:38:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:39:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:43:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:44:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:45:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:50:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:51:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:55:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:56:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:57:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
base_gcmessages.proto:61:18: Explicit 'optional' labels are disallowed in the Proto3 syntax. To define 'optional' fields in Proto3, simply remove the 'optional' label, as fields are 'optional' by default.
...
```

I believe those protos were meant to be compiled in protoc 2.X. I could try to convert them to 3.X, but it's too much.

**Comment by xfxyjwf** [comment]
Can you add syntax = "proto2" instead? 3.x supports both proto2 syntax and proto3 syntax.

**Comment by xfxyjwf** [comment]
Note that adding "syntax" statement is not required. You will see some warnings but it will still work.

**Comment by potato20** [comment]
I found the issue.

When I was installing `protoc` and did `sudo make install`, the permissions on `/usr/local/include/google` were 700 for the root user for some reason and the error messages weren't spitting that out for some reason. I decided to fix the permissions, re-installed protoc 2.6.1, then did

`protoc --proto_path=/usr/local/include  --proto_path=./csgo --ruby_out=./outlib csgo/**/*.proto`

with a fresh copy of the proto files (no syntax statements) and it worked.

Btw if I didn't specifiy `--proto_path=/usr/local/include`, even with permissions fixed, it fails with this

```
google/protobuf/descriptor.proto: File not found.
steammessages.proto: Import "google/protobuf/descriptor.proto" was not found or had errors.
steammessages.proto:6:8: ".google.protobuf.FieldOptions" is not defined.
steammessages.proto:10:8: ".google.protobuf.MessageOptions" is not defined.
steammessages.proto: ".google.protobuf.MessageOptions" is not defined.
base_gcmessages.proto: Import "steammessages.proto" was not found or had errors.
```

So why isn't `/usr/local/include/` included by default? Did I install protoc incorrectly? I more or less followed [this](https://gist.github.com/samklr/0b8a0620f82005e7f556) guide.

**Comment by xfxyjwf** [comment]
/usr/local/include is included by default in 3.x, so if you are using the latest version of protoc, you should not need either --proto_path=/usr/local/include or --proto_path=./.
